pkg/operator/ceph/controller/spec.go

```diff
@@ -264,20 +284,24 @@ func DaemonVolumesContainsPVC(volumes []v1.Volume) bool {
 
 // DaemonVolumes returns the pod volumes used by all Ceph daemons. If keyring resource name is
 // empty, there will be no keyring volume created from a secret.
-func DaemonVolumes(dataPaths *config.DataPathMap, keyringResourceName string) []v1.Volume {
-	vols := DaemonVolumesBase(dataPaths, keyringResourceName)
+func DaemonVolumes(dataPaths *config.DataPathMap, keyringResourceName string, dataDirHostPath string, isExporter bool) []v1.Volume {
+	vols := DaemonVolumesBase(dataPaths, keyringResourceName, dataDirHostPath, isExporter)
 	vols = append(vols, DaemonVolumesDataHostPath(dataPaths)...)
 	return vols
 }
 
 // DaemonVolumeMounts returns volume mounts which correspond to the DaemonVolumes. These
 // volume mounts are shared by most all Ceph daemon containers, both init and standard. If keyring
 // resource name is empty, there will be no keyring mounted in the container.
-func DaemonVolumeMounts(dataPaths *config.DataPathMap, keyringResourceName string) []v1.VolumeMount {
+func DaemonVolumeMounts(dataPaths *config.DataPathMap, keyringResourceName string, isExporter bool) []v1.VolumeMount {
```

Comment:
Instead of modifying this method signature with `isExporter`, how about if we revert this change? Then only in a single place where the exporter is created, it can append the volume mount after it calls this DaemonVolumeMounts() helper. This way all of the callers don't need to pass the argument.